src/base/Attestation.sol

```diff
@@ -62,6 +62,19 @@ abstract contract Attestation is IAttestation, AttestationResolve, ReentrancyGua
         ModuleRecord storage moduleRecord = _getModule({ moduleAddress: request.data.subject });
         ResolverUID resolverUID = moduleRecord.resolverUID;
 
+        // only run this function if the selected schemaUID exists
+        SchemaRecord storage schema = _getSchema({ schemaUID: request.schemaUID });
+        if (schema.registeredAt == ZERO_TIMESTAMP) revert InvalidSchema();
+        // validate Schema
+        ISchemaValidator validator = schema.validator;
+        // if validator is set, call the validator
+        if (address(validator) != ZERO_ADDRESS) {
+            // revert if ISchemaValidator returns false
+            if (!schema.validator.validateSchema(requestData)) {
+                revert InvalidAttestation();
+            }
+        }
+
         // write attestations to registry storge
```

Comment:
I think it would be good to implement this in an internal function, so we can reuse the code in AttestationDelegation